web-portal/components/DateTimePicker.vue

```diff
@@ -263,14 +263,16 @@
         this.edit_mode = true
         this.time_segment_index = which_segment
         const time_segment = this.value[which_segment]
-        this.picker = moment(time_segment.start_time).format('YYYY-MM-DD')
-        this.start_hour = moment(time_segment.start_time).format('hh')
-        this.start_minute = moment(time_segment.start_time).format('mm')
-        this.start_ampm = moment(time_segment.start_time).format('a')
-
-        this.end_hour = moment(time_segment.end_time).format('hh')
-        this.end_minute = moment(time_segment.end_time).format('mm')
-        this.end_ampm = moment(time_segment.end_time).format('a')
+        this.event_timezone = time_segment.timezone
+        const start_time = momenttz(time_segment.start_time).tz(this.event_timezone)
```

Comment:
small style suggestion, this might be easier to read with some new lines. Exactly where they go is subjective. I  usually try to split on related units. In this case I might stick new lines above the start_time and end_time `const` declarations